core/remote/watcher.js

```diff
@@ -109,6 +110,7 @@ class RemoteWatcher {
         throw userActionRequired.includeJSONintoError(err)
       } else {
         log.error({err})
+        this.events.emit('offline')
```

Comment:
There's a number of other errors we could get besides an unreachable host. What about emitting this event only if the error status is `0`?